sigma/processing/transformations.py

```diff
@@ -299,6 +301,163 @@ def apply_value(
         """
 
 
+class HashesFieldsDetectionItemTransformation(DetectionItemTransformation):
+    """
+    Transforms the 'Hashes' field in Sigma rules by creating separate detection items for each hash type.
+
+    This transformation replaces the generic 'Hashes' field with specific fields for each hash algorithm,
+    optionally prefixing the field names. It supports various hash formats and can auto-detect hash types
+    based on their length.
+
+    Attributes:
+        valid_hash_algos (List[str]): List of supported hash algorithms.
+        field_prefix (str): Prefix to add to the new field names.
+        drop_algo_prefix (bool): If True, omits the algorithm name from the new field name.
+        hash_lengths (Dict[int, str]): Mapping of hash lengths to their corresponding algorithms.
+
+    Example:
+        Input:
+            Hashes:
+                - 'SHA1=5F1CBC3D99558307BC1250D084FA968521482025'
+                - 'MD5=987B65CD9B9F4E9A1AFD8F8B48CF64A7'
+        Output:
+            FileSHA1: '5F1CBC3D99558307BC1250D084FA968521482025'
+            FileMD5: '987B65CD9B9F4E9A1AFD8F8B48CF64A7'
+    """
+
+    def __init__(
+        self, valid_hash_algos: List[str], field_prefix: str = None, drop_algo_prefix: bool = False
+    ):
+        """
+        Initializes the HashesDetectionItemTransformation.
+
+        Args:
+            valid_hash_algos (List[str]): List of valid hash algorithms supported by the pipeline/backend.
+            field_prefix (str, optional): Prefix to use for the new detection items. Defaults to None.
```

Comment:
I suggest to make the whole class a data class and move the whole initialization to the class-level declaration.